[text structure] Find field and message structure endpoints (#105660)
elastic/elasticsearch: docs/reference/text-structure/apis/find-structure-shared.asciidoc

`windows-1252`, or `EUC-JP`. If this parameter is not specified, the structure
finder chooses an appropriate character set.
end::param-charset[]

tag::param-column-names[]
`column_names`::
(Optional, string) If you have set `format` to `delimited`, you can specify the
column names in a comma-separated list. If this parameter is not specified, the
structure finder uses the column names from the header row of the text. If the
text does not have a header role, columns are named "column1", "column2",
"column3", etc.
end::param-column-names[]

tag::param-delimiter[]
`delimiter`::
(Optional, string) If you have set `format` to `delimited`, you can specify the
character used to delimit the values in each row. Only a single character is
supported; the delimiter cannot have multiple characters. By default, the API
considers the following possibilities: comma, tab, semi-colon, and pipe (`|`).
In this default scenario, all rows must have the same number of fields for the
delimited format to be detected. If you specify a delimiter, up to 10% of the
rows can have a different number of columns than the first row.
end::param-delimiter[]

tag::param-explain[]
`explain`::

> (Optional, Boolean) If this parameter is set to `true`, the response includes a
> field named `explanation`, which is an array of strings that indicate how the
> structure finder produced its result. The default value is `false`.

Reviewer: ```suggestion
(Optional, Boolean) If `true`, the response includes a
field named `explanation`, which is an array of strings that indicate how the
structure finder produced its result. The default value is `false`.
```

than would be seen with a bigger sample. If possible, however, it is more
efficient to upload sample text with more variety in the first 1000 lines than
to request analysis of 100000 lines to achieve some variety.

--
end::param-lines-to-sample[]

tag::param-quote[]
`quote`::
(Optional, string) If you have set `format` to `delimited`, you can specify the
character used to quote the values in each row if they contain newlines or the
delimiter character. Only a single character is supported. If this parameter is
not specified, the default value is a double quote (`"`). If your delimited text
format does not use quoting, a workaround is to set this argument to a character
that does not appear anywhere in the sample.
end::param-quote[]

tag::param-should-trim-fields[]
`should_trim_fields`::
(Optional, Boolean) If you have set `format` to `delimited`, you can specify
whether values between delimiters should have whitespace trimmed from them. If
this parameter is not specified and the delimiter is pipe (`|`), the default
value is `true`. Otherwise, the default value is `false`.
end::param-should-trim-fields[]

tag::param-timeout[]
`timeout`::

> (Optional, <<time-units,time units>>) Sets the maximum amount of time that the
> structure analysis make take. If the analysis is still running when the timeout
> expires then it will be aborted. The default value is 25 seconds.

Reviewer: ```suggestion
(Optional, <<time-units,time units>>) Sets the maximum amount of time that the
structure analysis may take. If the analysis is still running when the timeout
expires then it will be stopped. The default value is 25 seconds.
```

newline and carriage return, together with literal text enclosed in single
quotes. For example, `MM/dd HH.mm.ss,SSSSSS 'in' yyyy` is a valid override
format.

One valuable use case for this parameter is when the format is semi-structured
text, there are multiple timestamp formats in the text, and you know which
format corresponds to the primary timestamp, but you do not want to specify the
full `grok_pattern`. Another is when the timestamp format is one that the
structure finder does not consider by default.

If this parameter is not specified, the structure finder chooses the best
format from a built-in set.

If the special value `null` is specified the structure finder will not look
for a primary timestamp in the text. When the format is semi-structured text
this will result in the structure finder treating the text as single-line
messages.

The following table provides the appropriate `timeformat` values for some example timestamps:

|===
| Timeformat                 | Presentation

| yyyy-MM-dd HH:mm:ssZ       | 2019-04-20 13:15:22+0000
| EEE, d MMM yyyy HH:mm:ss Z | Sat, 20 Apr 2019 13:15:22 +0000
| dd.MM.yy HH:mm:ss.SSS      | 20.04.19 13:15:22.285
|===

> See

Reviewer: ```suggestion
Refer to
```
